src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +32,14 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream
+                .range(0, numbers.size())
+                .map(i -> i % 2 != 0 ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(n -> n % 2 != 0)
+                .mapToDouble(Double::valueOf)
+                .average()
```

Comment:
```suggestion
                .filter(n -> n % 2 != 0)                
                .average()
```
you dont need map it to double. Average function will do it